internal/riverinternaltest/riverdrivertest/riverdrivertest.go

```diff
@@ -0,0 +1,1731 @@
+			})
+
+			// Job is scheduled a while from now so it's not found.
+			jobRows, err := exec.JobGetAvailable(ctx, &riverdriver.JobGetAvailableParams{
+				AttemptedBy: clientID,
+				Max:         100,
+				Queue:       rivercommon.QueueDefault,
+			})
+			require.NoError(t, err)
+			require.Empty(t, jobRows)
+		})
+
+		t.Run("Prioritized", func(t *testing.T) {
+			t.Parallel()
+
+			exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+			// Insert jobs with decreasing priority numbers (3, 2, 1) which means increasing priority.
+			for i := 3; i > 0; i-- {
+				_ = testfactory.Job(ctx, t, exec, &testfactory.JobOpts{
+					Priority: &i,
+				})
+			}
+
+			jobRows, err := exec.JobGetAvailable(ctx, &riverdriver.JobGetAvailableParams{
+				AttemptedBy: clientID,
+				Max:         2,
+				Queue:       rivercommon.QueueDefault,
+			})
+			require.NoError(t, err)
+			require.Len(t, jobRows, 2, "expected to fetch exactly 2 jobs")
+
+			// Because the jobs are ordered within the fetch query's CTE but *not* within
+			// the final query, the final result list may not actually be sorted. This is
+			// fine, because we've already ensured that we've fetched the jobs we wanted
+			// to fetch via that ORDER BY. For testing we'll need to sort the list after
+			// fetch to easily assert that the expected jobs are in it.
+			sort.Slice(jobRows, func(i, j int) bool { return jobRows[i].Priority < jobRows[j].Priority })
+
+			require.Equal(t, 1, jobRows[0].Priority, "expected first job to have priority 1")
+			require.Equal(t, 2, jobRows[1].Priority, "expected second job to have priority 2")
+
+			// Should fetch the one remaining job on the next attempt:
+			jobRows, err = exec.JobGetAvailable(ctx, &riverdriver.JobGetAvailableParams{
+				AttemptedBy: clientID,
+				Max:         1,
+				Queue:       rivercommon.QueueDefault,
+			})
+			require.NoError(t, err)
+			require.NoError(t, err)
+			require.Len(t, jobRows, 1, "expected to fetch exactly 1 job")
+			require.Equal(t, 3, jobRows[0].Priority, "expected final job to have priority 3")
+		})
+	})
+
+	t.Run("JobGetByID", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("FetchesAnExistingJob", func(t *testing.T) {
+			t.Parallel()
+
+			exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+			now := time.Now().UTC()
+
+			job := testfactory.Job(ctx, t, exec, &testfactory.JobOpts{})
+
+			fetchedJob, err := exec.JobGetByID(ctx, job.ID)
+			require.NoError(t, err)
+			require.NotNil(t, fetchedJob)
+
+			require.Equal(t, job.ID, fetchedJob.ID)
+			require.Equal(t, rivertype.JobStateAvailable, fetchedJob.State)
+			require.WithinDuration(t, now, fetchedJob.CreatedAt, 100*time.Millisecond)
+			require.WithinDuration(t, now, fetchedJob.ScheduledAt, 100*time.Millisecond)
+		})
+
+		t.Run("ReturnsErrNotFoundIfJobDoesNotExist", func(t *testing.T) {
+			t.Parallel()
+
+			exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+			job, err := exec.JobGetByID(ctx, 99999)
```

Comment:
realized from some of our other tests that we can just use 0 here and have no possibility of a flaky test ever
```suggestion
			job, err := exec.JobGetByID(ctx, 0)
```

Reply:
Sweet fixed. There were a total of four instances of 9999 lookups between the riverdrivertest stuff I added and client_test. Updated all of them.